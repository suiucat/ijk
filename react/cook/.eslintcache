[{"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\App.js":"1","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\index.js":"2","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\reportWebVitals.js":"3","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\index.js":"4","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Button\\index.js":"5","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Table\\index.js":"6","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Search\\index.js":"7","C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\constants\\index.js":"8"},{"size":3330,"mtime":1606724495985,"results":"9","hashOfConfig":"10"},{"size":1430,"mtime":1606380395788,"results":"11","hashOfConfig":"10"},{"size":362,"mtime":1606185323803,"results":"12","hashOfConfig":"10"},{"size":500,"mtime":1606185323803,"results":"13","hashOfConfig":"10"},{"size":430,"mtime":1606723520853,"results":"14","hashOfConfig":"10"},{"size":1181,"mtime":1606724301024,"results":"15","hashOfConfig":"10"},{"size":291,"mtime":1606723198377,"results":"16","hashOfConfig":"10"},{"size":364,"mtime":1606723413694,"results":"17","hashOfConfig":"10"},{"filePath":"18","messages":"19","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1pyjwl9",{"filePath":"20","messages":"21","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"22","messages":"23","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"24","messages":"25","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"26"},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"29","messages":"30","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"31","messages":"32","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"33","messages":"34","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\App.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\index.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\reportWebVitals.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\index.js",[],["35","36"],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Button\\index.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Table\\index.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\components\\Search\\index.js",[],"C:\\Users\\benq\\Desktop\\demo\\react\\ijk\\react\\cook\\src\\constants\\index.js",[],{"ruleId":"37","replacedBy":"38"},{"ruleId":"39","replacedBy":"40"},"no-native-reassign",["41"],"no-negated-in-lhs",["42"],"no-global-assign","no-unsafe-negation"]