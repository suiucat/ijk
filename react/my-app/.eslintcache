[{"C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\reportWebVitals.js":"1","C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\App.js":"2","C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\index.js":"3"},{"size":362,"mtime":1606185323803,"results":"4","hashOfConfig":"5"},{"size":2219,"mtime":1606214547192,"results":"6","hashOfConfig":"5"},{"size":572,"mtime":1606188963425,"results":"7","hashOfConfig":"5"},{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1ue1yf5",{"filePath":"10","messages":"11","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\reportWebVitals.js",[],"C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\App.js",[],"C:\\Users\\benq\\Desktop\\demo\\learnreact\\my-app\\src\\index.js",[]]