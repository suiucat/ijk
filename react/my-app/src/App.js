import { example } from './components/example';

function App() {
  return (
    <div>
      <example></example>
    </div>
  );
}

export default App;
