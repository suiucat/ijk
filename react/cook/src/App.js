import { Component } from 'react';

import { Search, Table } from './components/index'
import './App.css';

const DEFAULT_QUERY = 'redux';
const DEFAULT_HPP = '100';
// const OPEN_LAZY_LOADING = true;

// 滚动条到底部懒加载
// scrollHeight: 可视内容 + 加上溢出的高度
// clientHeight: 页面可视区域的高度
// scrollTop: 滚动条在 Y 轴上的距离

// 判断滚动滚动到底部的条件： scrollHeight = clientHeight +  scrollTop
// okay 开干！
function lazyLoading() {
  const scrollHeight = window.document.body.scrollHeight,
        clientHeight = window.document.body.clientHeight,
        scrollTop = window.document.body.scrollTop;
        console.log(scrollHeight === clientHeight + scrollTop, scrollHeight, clientHeight, scrollTop)
        // if (scrollHeight === clientHeight + scrollTop && scrollHeight === 0) {
        //   return false;
        // } else if (scrollHeight === clientHeight + scrollTop && scrollHeight !== 0) {
        //   return true;
        // }
}


const PATH_BASE = 'https://hn.algolia.com/api/v1';
const PATH_SEARCH = '/search';
const PARAM_SEARCH = 'query=';
const PARAM_PAGE = 'page=';
const PARAM_HPP = 'hitsPerPage='
// const url = `${PATH_BASE}?${PARAM_SEARCH}${DEFAULT_QUERY}`

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      result: null,
      searchTerm: DEFAULT_QUERY,
    }

    this.setSearchStories = (result) => this.setState({result})
    this.fetchSearchTopStories = (searchTerm, page = 0) => {
      fetch(`
        ${PATH_BASE}${PATH_SEARCH}?${PARAM_SEARCH}${searchTerm}&${PARAM_PAGE}${page}&${PARAM_HPP}${DEFAULT_HPP}&`)
        .then(res => res.json())
        .then((result => this.setState({result})))
        .catch(e => console.log(e));
    }
    this.onSearchChange = (e) => {
      this.setState({
        searchTerm: e.target.value
      })
    }

    this.onDismiss = (id) => {
      const isNotId = item => item.objectID !== id;
      const updatedList = this.state.result.hits.filter(isNotId);
      this.setState({
        result: {...this.state.result, hits: updatedList } // 和 Object.assign({}. args...)效果一样
      })
    }

    this.onSearchSubmit = (event) => {
      const { searchTerm } = this.state;
      this.fetchSearchTopStories(searchTerm);
      event.preventDefault();
    }

    this.setSearchTopStories = (result) => {
      const { hits, page } = result;
      const oldHits = page === 0
        ? []
        : this.state.result.hits

    const updatedHits = [
      ...oldHits,
      ...hits,
    ];
    // console.log(updatedHits.length, '测试');
    this.setState({
      result: {
        hits: updatedHits,
        page,
    }
    })
    }
  }
  render() {
    const { searchTerm, result } = this.state;
    const page = (result && result.page) || 0;

    window.onscroll = function() {
      let isInFooter = lazyLoading();
      if (isInFooter) {
        this.fetchSearchTopStories(searchTerm, page + 1);
      }
    }

    return (
      <div className="page">
        <div className="interactions">
          <Search
            value={searchTerm}
            onChange={this.onSearchChange}
            onSubmit={this.onSearchSubmit}
          >
            搜索
          </Search>
        </div>
        {
          result
          && <Table
              list={result.hits}
              onDismiss={this.onDismiss}
            />
        }
        <div className="interactions">
          <button onClick={() => this.fetchSearchTopStories(searchTerm, page + 1)}>
            更多
          </button>
        </div>
      </div>
    );
  }
  componentDidMount() {
    const { searchTerm } = this.state;
    this.fetchSearchTopStories(searchTerm);
  }

  // shouldComponentUpdate(nP, nS) {
  //   console.log('shouldComponentUpdate', 4, nP, nS);
  //   return true;
  // }

  // componentDidUpdate(pP,pS) {
  //   console.log('componentDidUpdate', 6, pP,pS);
  // }

  // componentDidCatch(err, info) {
  //   console.log('componentDidCatch', 8, err, info)
  // }
}

export default App;
