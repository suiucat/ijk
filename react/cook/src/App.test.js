import { render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import App, { Search, Table, Button} from './App';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('has a vaild snapshot', () => {
    const component = renderer.create(
      <App />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('Search', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
  });

  test('has a vaild snapshot', () => {
    const component = renderer.create(
      <Search />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('Button', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
  });

  test('has a vaild snapshot', () => {
    const component = renderer.create(
      <Button />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Table', () => {
  const props = {
    list: [{ title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
           { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
          ],
    sortKey: 'TITLE',
    isSortReverse: false,
    };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table { ...props }/>, div);
  });

  test('has a vaild snapshot', () => {
    const component = renderer.create(
      <Table { ...props }/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});