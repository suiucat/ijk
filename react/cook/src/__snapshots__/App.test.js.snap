// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` has a vaild snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`App has a vaild snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`App has a vaild snapshot 2`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`Button has a vaild snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;

exports[`Button has a vaild snapshot 2`] = `
<button
  className=""
  type="button"
/>
`;

exports[`Search has a vaild snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;

exports[`Search has a vaild snapshot 2`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;

exports[`Table has a vaild snapshot 1`] = `
<button
  className=""
  type="button"
/>
`;
